packages/jest-worker/src/base/BaseWorkerPool.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+'use strict';
+
+import mergeStream from 'merge-stream';
+import os from 'os';
+import path from 'path';
+
+import ChildProcessWorker from '../workers/ChildProcessWorker';
+import NodeThreadsWorker from '../workers/NodeThreadsWorker';
+import {CHILD_MESSAGE_END} from '../types';
+
+import type {Readable} from 'stream';
+import type {FarmOptions, WorkerOptions, WorkerInterface} from '../types';
+
+/* istanbul ignore next */
+const emptyMethod = () => {};
+
+export default class BaseWorkerPool {
+  _stderr: Readable;
+  _stdout: Readable;
+  _options: FarmOptions;
+  _workers: Array<WorkerInterface>;
+
+  constructor(workerPath: string, options: FarmOptions) {
+    this._options = options;
+
+    const numWorkers = options.numWorkers || os.cpus().length - 1;
+    this._workers = new Array(numWorkers);
+
+    if (!path.isAbsolute(workerPath)) {
+      workerPath = require.resolve(workerPath);
+    }
+
+    const stdout = mergeStream();
+    const stderr = mergeStream();
+
+    for (let i = 0; i < numWorkers; i++) {
+      const workerOptions: WorkerOptions = {
+        forkOptions: options.forkOptions || {},
+        maxRetries: options.maxRetries || 3,
+        useNodeWorkersIfPossible: options.useNodeWorkersIfPossible,
+        workerId: i,
+        workerPath,
+      };
+
+      const worker = this.createWorker(workerOptions);
+      const workerStdout = worker.getStdout();
+      const workerStderr = worker.getStderr();
+
+      if (workerStdout) {
+        stdout.add(workerStdout);
+      }
+
+      if (workerStderr) {
+        stderr.add(workerStderr);
+      }
+
+      this._workers[i] = worker;
+
+      // $FlowFixMe
```

Comment:
Please add explanation to all `FlowFixMe`s.

You can get rid of this ones by redefining at the top the result, as a variable (class methods are covariant in Flow). For instance, putting at the top: `getStderr: () => BaseWorkerPool;` would do the trick.

Anyway, I'm not sure why do we need to `bind` them all.

Reply:
We actually don't need it anymore. Will remove them accordingly